src/main/java/nu/studer/teamcity/buildscan/internal/cleanup/BuildScanAdminPage.java

```diff
@@ -0,0 +1,42 @@
+package nu.studer.teamcity.buildscan.internal.cleanup;
+
+import jetbrains.buildServer.controllers.admin.AdminPage;
+import jetbrains.buildServer.serverSide.auth.Permission;
+import jetbrains.buildServer.web.openapi.PagePlaces;
+import jetbrains.buildServer.web.openapi.PluginDescriptor;
+import jetbrains.buildServer.web.openapi.PositionConstraint;
+import org.jetbrains.annotations.NotNull;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Map;
+
+public class BuildScanAdminPage extends AdminPage {
+
+    private final CustomDataStorageBuildScanCleaner customDataStorageBuildScanCleaner;
+
+    public BuildScanAdminPage(@NotNull PagePlaces pagePlaces, @NotNull PluginDescriptor descriptor, CustomDataStorageBuildScanCleaner customDataStorageBuildScanCleaner) {
+        super(pagePlaces);
+        this.customDataStorageBuildScanCleaner = customDataStorageBuildScanCleaner;
+        setPluginName(descriptor.getPluginName());
+        setIncludeUrl(descriptor.getPluginResourcesPath("/admin/buildScanAdminPage.jsp"));
+        addJsFile(descriptor.getPluginResourcesPath("js/buildScan.js"));
+        setTabTitle("Build Scan");
+        setPosition(PositionConstraint.last());
+        register();
```

Comment:
I suggest to replace this with the `init` configuration on the bean in the Spring XML file.

Reply:
Done